R/FilterStateChoices.R

```diff
@@ -163,15 +163,18 @@ ChoicesFilterState <- R6::R6Class( # nolint
       if (inherits(x, "POSIXt")) {
         private$tzone <- Find(function(x) x != "", attr(as.POSIXlt(x), "tzone"))
       }
+      private$set_is_choice_limited(x, choices)
+      x <- x[x %in% choices]
+      selected <- selected[selected %in% choices]
 
       if (!is.factor(x)) {
         x <- factor(as.character(x), levels = as.character(sort(unique(x))))
       }
       x <- droplevels(x)
-      choices <- table(x)
-      private$set_choices(names(choices))
-      self$set_selected(names(choices))
-      private$set_choices_counts(unname(choices))
+      choices_table <- table(x)
+      private$set_choices(names(choices_table))
+      self$set_selected(selected)
```

Comment:
Since super$initialize gets `choices` and `selected`, these calls doesn't need to be here as set_choices and set_selected is done twice. This means that `super$initialize` should at the end this init.

Reply:
I see that choices are not yet used in `super$initialize`, and selected is set as `private$selected <- reactiveVal(NULL)`. So we can skip this line, adjust choices and call `super$initialize` at the end, right? And same goes for other types?

Reply:
If `FilterState$initialize` has argument `choices` and `selected` but doesn't use them then it's wrong

Reply:
`set_selected` in all subclasses calls `super$set_selected`. It means you can easily remove them all. Then it might be clear that `set_selected` is an abstract process and probably should be moved to the abstract class.

https://github.com/insightsengineering/teal.slice/blob/3ee6bdd60cac3b8db5c497c621127226ec1acd4c/R/FilterStateRange.R#L323